server_environment/controllers/main.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2019 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from werkzeug.utils import redirect
+from urllib.parse import urljoin
+
+from odoo.tools.config import config
+from odoo import http
+from odoo.http import request, Controller
+
+
+class ServerEnvironmentController(Controller):
+
+    @http.route(
+        '/server_environment_<module_extension>'
+        '/static/RUNNING_ENV/<local_path>',
+        type='http', auth='public')
+    def environment_redirect(self, module_extension, local_path, **kw):
+        # Note: module_extension is present to make working
+        # the module in normal configuration, with the folder
+        # server_environment_files and in demo configuration, with the
+        # module  server_environment_files_sample
+        running_env = config.get('running_env', "default")
+        IrConfigParameter = request.env['ir.config_parameter'].sudo()
+        base_url = IrConfigParameter.get_param('web.base.url', '')
+        new_path = "/server_environment_%s/static/%s/%s" % (
```

Comment:
if the env has no file this will give 404 every time you hit it.
Imagine you want to change styles only to make prominent that you are on test instead of prod.
On prod you'll get tons of 404, unless you recommend to always have a file for all envs (?).

Reply:
>  On prod you'll get tons of 404, unless you recommend to always have a file for all envs (?).

Exactly. As for the other part of this module. you should have a folder for each environment, event if it's empty.

Reply:
Note : it is written in the ``CONFIGURE.rst`` file : 

```
Then, create css files for each environment you have. exemple:


    /server_environment_files/static/dev/css.css
    /server_environment_files/static/prod/css.css
    ...
```